repo/blob/azure/azure_storage.go

```diff
@@ -133,14 +133,21 @@ func translateError(err error) error {
 }
 
 func (az *azStorage) PutBlob(ctx context.Context, b blob.ID, data blob.Bytes, opts blob.PutOptions) error {
-	switch {
-	case opts.HasRetentionOptions():
-		return errors.Wrap(blob.ErrUnsupportedPutBlobOption, "blob-retention")
-	case opts.DoNotRecreate:
+	if opts.DoNotRecreate {
 		return errors.Wrap(blob.ErrUnsupportedPutBlobOption, "do-not-recreate")
 	}
 
-	_, err := az.putBlob(ctx, b, data, opts)
+	hardcodedOpts := blob.PutOptions{
```

Comment:
nit: not sure about the variable name.
Anyway, merging now and we'll modify it separately.